frontend/src/components/common/Terminal.tsx

```diff
@@ -197,8 +197,30 @@ export default function Terminal(props: TerminalProps) {
       shellConnectFailed(xtermc);
       return;
     }
-
-    xterm.write(text);
+    // replace \n with \r only if there is \r before \n
+    // give me the regular expression for the above
+    text.replace(/\r\n/g, '\r').replace(/\n/g, '\r');
+    if (props.isAttach) {
+      // in case of attach if we didn't recieve any data from the process we should try to reconnect
+      if (!text) {
+        text = "If you don't see a command prompt, try pressing enter.";
+      }
+      // in case of attach we should replace \n with \r\n because it's not formatted the way it is for exec so we have to manually do this
+      xterm.write(
+        text.replace(/(\r)?\n/g, function (match, p1) {
+          // If there is a \r before \n, replace \n with \r\n
+          if (p1 === '\r') {
+            return '';
+          }
+          // Otherwise, just return \r\n
+          else {
+            return '\r\n';
+          }
+        })
```

Comment:
I think this whole regex can be simplified by doing:
`text = text.replace(/\r\n/g, '\n').replace(/\n/g, '\r\n');`

Can you check if this is the case?